pygments/plugin.py

```diff
@@ -34,19 +36,34 @@
     :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
     :license: BSD, see LICENSE for details.
 """
+
+from warnings import filterwarnings
+
 LEXER_ENTRY_POINT = 'pygments.lexers'
 FORMATTER_ENTRY_POINT = 'pygments.formatters'
 STYLE_ENTRY_POINT = 'pygments.styles'
 FILTER_ENTRY_POINT = 'pygments.filters'
 
 
 def iter_entry_points(group_name):
+    # We can use .select() when we no longer support Python 3.9.
+    filterwarnings('ignore', 'SelectableGroups dict interface is deprecated. Use select.', DeprecationWarning)
     try:
-        import pkg_resources
-    except (ImportError, OSError):
-        return []
-
-    return pkg_resources.iter_entry_points(group_name)
+        from importlib.metadata import entry_points
+    except ImportError:
+        try:
+            from importlib_metadata import entry_points
+        except ImportError:
+            try:
+                from pkg_resources import iter_entry_points
+            except (ImportError, OSError):
```

Comment:
Here I have a question. I kept this from the old code. Is there any particular reason an import might trigger `OSError`?

Reply:
I don't remember anything... possibly `pkg_resources` does some things already on import which might trigger this?

Reply:
Ok, then I’ll stick with ImportError for the others, but keep OSError here to minimize risk.